File: 169-MajorityElement/169-MajorityElement.py
# Last updated: 1/8/2026, 3:34:28 PM
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        s=set()
        n=len(nums)
        for i in nums:
            s.add(i)
        for i in s:
            c=nums.count(i)
            if c>(n/2):
                return i

File: 169-MajorityElement/169-MajorityElement.java
// Last updated: 9/10/2026, 10:00:11 AM
class Solution {
    public int majorityElement(int[] nums) {
        int res=0;
        int count=0;
        for(int num:nums){
            if(count<=0){
                res=num;
            }
            if(res==num) count++;
            if(res!=num) count--;
        }
        return res;
    }
}